crates/wast/src/binary.rs

```diff
@@ -144,6 +153,110 @@ fn encode_fields(
     }
 }
 
+fn encode_component_fields(
+    component_id: &Option<Id<'_>>,
+    component_name: &Option<NameAnnotation<'_>>,
+    fields: &[ComponentField<'_>],
+) -> Vec<u8> {
+    use crate::ast::CustomPlace::*;
+    use crate::ast::CustomPlaceAnchor::*;
+
+    let mut types = Vec::new();
+    let mut imports = Vec::new();
+    let mut funcs = Vec::new();
+    let mut exports = Vec::new();
+    let mut start = Vec::new();
+    let mut customs = Vec::new();
+    let mut instances = Vec::new();
+    let mut modules = Vec::new();
+    let mut components = Vec::new();
+    let mut aliases = Vec::new();
+    for field in fields {
+        match field {
+            ComponentField::Type(i) => types.push(i),
+            ComponentField::Import(i) => imports.push(i),
+            ComponentField::Func(i) => funcs.push(i),
+            ComponentField::Export(i) => exports.push(i),
+            ComponentField::Start(i) => start.push(i),
+            ComponentField::Custom(i) => customs.push(i),
+            ComponentField::Instance(i) => instances.push(i),
+            ComponentField::NestedModule(i) => modules.push(i),
+            ComponentField::NestedComponent(i) => components.push(i),
+            ComponentField::Alias(a) => aliases.push(a),
+        }
+    }
```

Comment:
Similar to below, while this is done for core wasm modules I don't think this will be applicable for components.